diff --git a/config.def.h b/config.def.h
index 77ff358..67ed1f3 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,41 +1,45 @@
+#include <X11/XF86keysym.h>
+
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
 static const char font[]            = "-*-terminus-medium-r-*-*-16-*-*-*-*-*-*-*";
-static const char normbordercolor[] = "#444444";
-static const char normbgcolor[]     = "#222222";
-static const char normfgcolor[]     = "#bbbbbb";
-static const char selbordercolor[]  = "#005577";
-static const char selbgcolor[]      = "#005577";
-static const char selfgcolor[]      = "#eeeeee";
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static const char normbordercolor[] = "#202020";
+static const char normbgcolor[]     = "#202020";
+static const char normfgcolor[]     = "#999999";
+static const char selbordercolor[]  = "#202020";
+static const char selbgcolor[]      = "#202020";
+static const char selfgcolor[]      = "#eeeeec";
+static const unsigned int borderpx  = 0;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const Bool showbar           = True;     /* False means no bar */
 static const Bool topbar            = True;     /* False means bottom bar */
+static const Bool viewontag         = True;     /* Switch view on tag switch */
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "trm", "web", "rdp", "a", "b", "c", "d", "e", "f" };
 
 static const Rule rules[] = {
-	/* class      instance    title       tags mask     isfloating   monitor */
-	{ "Gimp",     NULL,       NULL,       0,            True,        -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       False,       -1 },
+	/* class			instance    	title	tags mask	isfloating	monitor */
+	{ "st-256color",    "st-256color",  NULL,   1,          False,      -1 },
+	{ "Firefox",   		NULL,  			NULL, 	1 << 1,     False,      -1 },
+	{ "xfreerdp",  		"xfreerdp",		NULL,	1 << 3,     False,      -1 },
 };
 
 /* layout(s) */
 static const float mfact      = 0.55; /* factor of master area size [0.05..0.95] */
 static const int nmaster      = 1;    /* number of clients in master area */
-static const Bool resizehints = True; /* True means respect size hints in tiled resizals */
+static const Bool resizehints = False; /* True means respect size hints in tiled resizals */
 
 static const Layout layouts[] = {
 	/* symbol     arrange function */
-	{ "[]=",      tile },    /* first entry is default */
-	{ "><>",      NULL },    /* no layout function means floating behavior */
-	{ "[M]",      monocle },
+	{ "#",      tile },    /* first entry is default */
+	{ "~",      NULL },    /* no layout function means floating behavior */
+	{ "|",      monocle },
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -47,10 +51,16 @@ static const Layout layouts[] = {
 
 /* commands */
 static const char *dmenucmd[] = { "dmenu_run", "-fn", font, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbgcolor, "-sf", selfgcolor, NULL };
-static const char *termcmd[]  = { "uxterm", NULL };
+static const char *termcmd[]  = { "st", NULL };
+static const char *browsercmd[] = { "firefox", NULL };
+static const char *volraisecmd[] = { "amixer", "-c", "1", "set", "Master", "1+", "unmute",  NULL };
+static const char *vollowercmd[] = { "amixer", "-c", "1", "set", "Master", "1-", "unmute",  NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
+	{ 0,         XF86XK_AudioRaiseVolume,      spawn,          {.v = volraisecmd } },
+	{ 0,         XF86XK_AudioLowerVolume,      spawn,          {.v = vollowercmd } },
+	{ MODKEY|ShiftMask,             XK_b,      spawn,          {.v = browsercmd } },
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
diff --git a/dwm.c b/dwm.c
index 1d78655..65208a0 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1680,6 +1680,8 @@ tag(const Arg *arg) {
 		selmon->sel->tags = arg->ui & TAGMASK;
 		focus(NULL);
 		arrange(selmon);
+		if(viewontag)
+			view(arg);
 	}
 }
 
